Add Docker Hub automated build hook

hanachin opened the pull request:
I take docker logo from Docker Media Kit
https://www.docker.com/brand-guidelines

hibariya commented on lib/hooks/docker-hub-automated-build/help.html.haml:
```diff
@@ -0,0 +1,4 @@
+  %dl
```
This indentation seems causing CI failure (`Haml::SyntaxError: Indenting at the beginning of the document is illegal.`). Would you please remove this?

hanachin replied:
Thanks, I removed indentation and force pushed the commit.
By the way, I tested this hook by following "Preview your hook" section in the README, I want to know how can I preview the help.html.haml? is it possible?
https://github.com/idobata/idobata-hooks#preview-your-hook

hibariya replied:
Thanks!

AFAIK, we can't preview the `help.html.haml` with that.
However, you can confirm the rendered html (hbs) on `app/templates/idobata-hooks/HOOK_NAME/help.hbs` by running `yarn run prepublish`. Unfortunately, probably due to an issue of `hamljs`, it results syntax error on this branch though💦 I'll investigate it later.

hanachin replied:
Thank you for letting me know 👍

hibariya commented on lib/hooks/docker-hub-automated-build/templates/default.html.haml:
```diff
@@ -0,0 +1,4 @@
+%div
+  #{payload.push_data.pusher} pushed #{payload.repository.repo_name}:#{payload.push_data.tag}
+  %br
+  #{payload.repository.repo_url}
```
How about using this URL for the link of repo name (`#{payload.repository.repo_name}`) above?  In that way, we can remove this 2nd line.

hanachin replied:
I changed the template to link `repo_name` to the repository url, It looks nicer now. thanks!

![2017-06-05 18 41 06](https://cloud.githubusercontent.com/assets/167012/26779117/236ca51e-4a1f-11e7-9b84-d35a47b1e54a.png)

hanachin commented on lib/hooks/docker-hub-automated-build/hook.rb:
```diff
@@ -0,0 +1,6 @@
+module Idobata::Hook
+  class DockerHubAutomatedBuild < Base
```
Ok, I'll rename it.

hibariya commented on lib/hooks/docker-hub/hook.rb:
```diff
@@ -0,0 +1,6 @@
+module Idobata::Hook
+  class DockerHub < Base
+    screen_name 'Docker Hub automated build'
```
I think 'Docker Hub' is more appropriate for now.

hanachin replied:
Ooops, I missed it 🙈 I fixed in 46361e0

hibariya commented:
Thank you for your quick response!

I'm sorry many times :pray: I've totally forgotten about tests.  Could you add a test that demonstrate this hook treats a payload from Docker Hub appropriately?

I scaled the icon image by resizing canvas size so that it can fit appropriately in Idobata. I believe this change respects the guidelines.
![mono_vertical_small_square](https://cloud.githubusercontent.com/assets/43346/26811606/e67c852c-4aad-11e7-8af6-5abd4615b104.png)

hanachin commented:
Thanks for all your help 👍 Please give me a little more time, I'll get real docker hub automated build webhook then write a test.

> I believe this change respects the guidelines.

I believe it too. I'll add a resized version of the icon later.

hanachin commented:
I use requestb.in to get docker hub webhook request.
https://requestb.in/xye1l5xy?inspect

I added the test in 81d7eb8, and added resized logo image in f617556 😃

hibariya commented:
@hanachin  Thank you! I'll deploy this new hook to idobata.io as soon as I can :rocket:
